src/tests/functional/shared_test_classes/src/subgraph/matmul_split_decompose.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "shared_test_classes/subgraph/matmul_split_decompose.hpp"
+
+#include "common_test_utils/graph_comparator.hpp"
+#include "functional_test_utils/skip_tests_config.hpp"
+#include "openvino/pass/manager.hpp"
+#include "openvino/runtime/exec_model_info.hpp"
+#include "transformations/common_optimizations/matmul_split_decomposition.hpp"
+#include "common_test_utils/data_utils.hpp"
+
+namespace ov {
+namespace test {
+
+inline void CheckNumberOfNodesWithType(std::shared_ptr<const ov::Model> function,
```

Comment:
We already have several cases when we need to check number of nodes with specific types in `SubgraphBaseTest`. Maybe we can move this function to `ov_subgraph.hpp` (as it is done in cpu_test_utils.hpp)?